src/aggregation/metric/top_hits.rs

```diff
@@ -0,0 +1,820 @@
+///           "docvalue_fields": {
+///             "date": "<date_RFC3339>",
+///             "title": "<title>",
+///             "iden": "<iden>"
+///           }
+///         }
+///     ]
+/// }
+/// ```
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Default)]
+pub struct TopHitsAggregation {
+    sort: Vec<KeyOrder>,
+    size: usize,
+    from: Option<usize>,
+
+    #[serde(flatten)]
+    retrieval: RetrievalFields,
+}
+
+const fn default_doc_value_fields() -> Vec<String> {
+    Vec::new()
+}
+
+/// Search query spec for each matched document
+/// TODO: move this to a common module
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Default)]
+pub struct RetrievalFields {
+    /// The fast fields to return for each hit.
+    /// This is the only variant supported for now.
+    /// TODO: support the {field, format} variant for custom formatting.
+    #[serde(rename = "docvalue_fields")]
+    #[serde(default = "default_doc_value_fields")]
+    pub doc_value_fields: Vec<String>,
+}
+
+/// Search query result for each matched document
+/// TODO: move this to a common module
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Default)]
+pub struct FieldRetrivalResult {
+    /// The fast fields returned for each hit.
+    #[serde(rename = "docvalue_fields")]
+    #[serde(skip_serializing_if = "HashMap::is_empty")]
+    pub doc_value_fields: HashMap<String, OwnedValue>,
+}
+
+impl RetrievalFields {
+    fn get_field_names(&self) -> Vec<&str> {
+        self.doc_value_fields.iter().map(|s| s.as_str()).collect()
+    }
+
+    fn resolve_field_names(self, reader: &ColumnarReader) -> crate::Result<Self> {
+        let globbed_string_to_regex = |globbed: &str| {
+            let sanitized = format!("^{}$", regex::escape(globbed).replace(r"\*", ".*"));
+            Regex::new(&sanitized.replace('*', ".*")).map_err(|e| {
+                crate::TantivyError::SchemaError(format!(
+                    "Invalid regex '{}' in docvalue_fields: {}",
+                    globbed, e
+                ))
+            })
+        };
+        let resolved_fields = self
+            .doc_value_fields
+            .into_iter()
+            .map(|field| {
+                if !field.contains('*') && reader.iter_columns()?.any(|(name, _)| name == field) {
+                    return Ok(vec![field]);
+                }
+
+                let pattern = globbed_string_to_regex(&field)?;
+                Ok(reader
+                    .iter_columns()?
+                    .filter(|(name, _)| pattern.is_match(name))
+                    .map(|(name, _)| name)
+                    .collect::<Vec<_>>())
+            })
+            .collect::<crate::Result<Vec<_>>>()?
+            .into_iter()
+            .flatten()
+            .collect();
+
+        Ok(Self {
+            doc_value_fields: resolved_fields,
+        })
+    }
+
+    fn get_fields(
+        &self,
+        accessors: &HashMap<String, DynamicColumn>,
```

Comment:
There may be multiple columns per column name, so it should be a
tuple `(String, ColumnType)`

Reply:
Just to confirm, result `docvalue_fields` should still be grouped by just name?
```jsonc
{
	"docvalue_fields": {
             // Corresponds to ctypes date and u64 for example
		"created": [
			"2023-07-08T...",
			091303333333
		]
	}
}
```

Reply:
See [33b12a8](https://github.com/quickwit-oss/tantivy/pull/2198/commits/33b12a8cf88d5e0034554a593432a60c3b8515b6)

Reply:
> Just to confirm, result `docvalue_fields` should still be grouped by just name?
> 
> ```js
> {
> 	"docvalue_fields": {
>              // Corresponds to ctypes date and u64 for example
> 		"created": [
> 			"2023-07-08T...",
> 			091303333333
> 		]
> 	}
> }
> ```

I think yes. We can't be 100% compatible with elastic search here, as they don't support mixed types.

Reply:
Fair, I've added the functionality to top_hits. I was considering covering this in the tests but `SchemaBuilder::add_field` does not allow duplicate field names, is there another way to create one?

Reply:
Ran into a blocker while writing the test. For nested JSON fields
1. `ColumnarReader` seems to remove the `.` separators from field names (as they exist in the `column_dictionary`.
2. `SegmentReader` retains the `.` separator.

What this means: Let's say we have a nested field `json.f` we first resolve field names from `ColumnarReader` and get back `jsonf` which we're then unable to query through the `SegmentReader` during the aggregations.

Is `1.` expected behavior? If yes, what could be an alternative way to resolve column names before the aggregation.